sequential/quadtree.cpp

```diff
@@ -0,0 +1,248 @@
+                                             Point(BottomRight.x, (TopLeft.y + BottomRight.y) / 2)
+                                          );
+            }
+
+            // Recursive insert call
+            TopRightTree -> insert(node);
+        }
+
+        else{ // Bottom Right Quarter
+            if(BottomRightTree == NULL) {
+                // Create the grid according to the coordinates of the parent grid
+                BottomRightTree = new QuadTree( Point ((TopLeft.x + BottomRight.x) / 2, 
+                                                       (TopLeft.y + BottomRight.y)/ 2),
+                                                Point (BottomRight.x, BottomRight.y)
+                                             );
+            }
+
+            // Recursive call
+            BottomRightTree -> insert(node);
+        }
+    }
+}
+
+// Function to find a Node in the QuadTree
+Node* QuadTree :: search(Point p) {
+    // Check if in boundary
+    if(!inBoundary(p))
+        return NULL;
+
+    // if the current node is not null, i.e. smallest grid, return the node value
+    if(n != NULL)
+        return n;
+    
+    // Check for coordinates of the point if they lie in first half or the second
+    if((TopLeft.x + BottomRight.x) / 2 >= p.x) { // In Left
+
+        // If in top Quarter
+        if((TopLeft.y + BottomRight.y) / 2 <= p.y) {
+            // Check if the grid is empty, return NULL
+            if(TopLeftTree == NULL)
+                return NULL;
+
+            return TopLeftTree -> search(p);
+        }
+        // If in bottom Quarter
+        else {
+            // Check if the grid is empty, return NULL
+            if(BottomLeftTree == NULL) 
+                return NULL;
+
+            return BottomLeftTree -> search(p);
+        }
+    }
+
+    else { // In Right
+
+        // If in top Quarter
+        if((TopLeft.y + BottomRight.y) / 2 <= p.y) {
+            // Check if the grid is empty, return NULL
+            if(TopRightTree == NULL)
+                return NULL;
+
+            return TopRightTree -> search(p);
+        }
+        // If in bottom Quarter
+        else {
+            if(BottomRightTree == NULL)
+                return NULL;
+
+            return BottomRightTree -> search(p);
+        }
+    }
+};
+
+// Function to parse the .txt file and return a vector of Points
+vector<pair<int,int>> parseCoordinates(const string& filename) {
+    vector<pair<int,int>> coordinates;
+    ifstream infile(filename);
+
+    if (!infile) {
+        cerr << "Error: Could not open file " << filename << endl;
+        exit(0);
+    }
+
+    string line;
+    while (getline(infile, line)) {
+        stringstream ss(line);
+        int x, y;
+        if (ss >> x >> y) {
+            coordinates.push_back({x, y});
+        }
+    }
+
+    infile.close();
+    return coordinates;
+}
+
+
+
+int main() {
+    // Random Grid
+    cout<<"START\n";
+    QuadTree* root = new QuadTree(Point(0, 1000), Point(1000, 0));
```

Comment:
Can we make the maximum boundary size a parameter that is fed into the file as a command line argument?

Reply:
Yes, will make the changes

Reply:
Yeah can do